cms/src/test/java/uk/nhs/digital/apispecs/IsAnyTrueHelperTest.java

```diff
@@ -0,0 +1,126 @@
+package uk.nhs.digital.apispecs;
+
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import com.github.jknack.handlebars.*;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import uk.nhs.digital.apispecs.swagger.IsAnyTrueHelper;
+
+import java.io.IOException;
+import java.util.*;
+
+public class IsAnyTrueHelperTest {
+
+    private static final String TEMPLATE_CONTENT_FROM_THE_POSITIVE_BLOCK = RandomStringUtils.random(10);
+    private static final String TEMPLATE_CONTENT_FROM_THE_INVERSE_BLOCK = RandomStringUtils.random(10);
+
+    private final IsAnyTrueHelper isAnyTrueHelper = IsAnyTrueHelper.INSTANCE;
+
+    @Mock private Options.Buffer buffer;
+    private String item = "some-item";
+
+    @Before
+    public void setUp() throws Exception {
+        initMocks(this);
+    }
+
+    @Test
+    public void rendersBlockWhenCollectionHasAtLeastOneItem() throws IOException {
+
+        Boolean[] params = {false,false,true,false,false};
+        MockOptions options = new MockOptions(params);
+
+        // when
+        final Object buffer = isAnyTrueHelper.apply(item, options);
+
+        // then
+        assertThat("Returns Options.Buffer",
+            buffer,
+            instanceOf(Options.Buffer.class)
+        );
+
+        final Options.Buffer actualBuffer = (Options.Buffer) buffer;
+
+        then(actualBuffer).should().append(TEMPLATE_CONTENT_FROM_THE_POSITIVE_BLOCK);
+    }
+
+    @Test
+    public void throwExceptionWhenCollectionIsEmpty() throws IOException {
+
+        Boolean[] params = {};
+        MockOptions options = new MockOptions(params);
+
+        try {
+            // when
+            isAnyTrueHelper.apply(item, options);
+
+        } catch (Exception e) {
+
+            // then
+            assertThat("Exception in execution","Exception in IsAnyTrueHelper", is(e.getMessage()));
+        }
+    }
+
+    @Test
+    public void throwExceptionWhenCollectionIsNull() throws IOException {
+
+        Boolean[] params = null;
+        MockOptions options = new MockOptions(params);
+
+        try {
+            // when
+            isAnyTrueHelper.apply(item, options);
+
+        } catch (Exception e) {
+
+            // then
+            assertThat("Exception in execution","Exception in IsAnyTrueHelper", is(e.getMessage()));
+        }
+    }
+
+    @Test
+    public void throwExceptionWhenCollectionIsNotBoolean() throws IOException {
```

Comment:
Will this test fail when the helper throws no exception?

Reply:
Fixed

Reply:
I'm afraid that if I make the helper throw no exception, this test still passes - and yet, it was meant to verify that we throw validation exception with a specific message.

Indeed, if I _only_ comment out `Validate.allElementsOfType` it does fail but it's not because _validation_ exception was thrown.

This test needs to assert that the exception thrown is of the type thrown by the `Validate` method it verifies  - otherwise it's not checking this specific validation. Also, it should fail when absolutely no exception is thrown from the method under test.

Reply:
fixed